### ISSUE TITLE
iconSize Inconstancies

### ISSUE THREAD

**Issue by stephenheron** [nitaliano/react-native-mapbox-gl#1417]
Hi,

I just upgraded to the latest version of the library and I ran into a little bit of a inconsistency between debug and release compared to the older version. The way the iconSize style on the SymbolLayer seems to behaving strangely. 

My icon is around 77x77 and then I was using iconSize: 0.35 to make the icon smaller. 

Now after upgrading I am seeing the following:

**iOS Debug:**
<img width="354" alt="ios-pin-dev" src="https://user-images.githubusercontent.com/142855/48094722-c821aa80-e20a-11e8-8ae3-efec7c6d3ecc.png">

**iOS Release**
<img width="350" alt="ios-pin-prod" src="https://user-images.githubusercontent.com/142855/48094739-d5d73000-e20a-11e8-8609-1336ca2ff15f.png">

**Android Debug**
<img width="256" alt="pin-dev" src="https://user-images.githubusercontent.com/142855/48094751-de2f6b00-e20a-11e8-8ad1-9d22fc429103.png">

**Android Release**
<img width="259" alt="pin-prod" src="https://user-images.githubusercontent.com/142855/48094761-e7b8d300-e20a-11e8-8c49-8d44144e49fb.png">

As you can see the size difference is quite different between debug and release. I was wondering if I am doing wrong with the iconSize or if it is a bug?

Anyone seeing something similar?

Thanks,
Stephen

**Comment by Darkilen** [comment]
Hello,

I've exactly the same issue (test on Android only for now).

**Android Release**
![android_release](https://user-images.githubusercontent.com/22028659/49294554-93281100-f4b3-11e8-8ab2-f267c61fd7a4.png)


**Android Debug**
![android_debug](https://user-images.githubusercontent.com/22028659/49294566-97ecc500-f4b3-11e8-8fae-1f1ed6496cbd.png)


```
Image size : 41x41
iconSize: 0.8
```
```
"@mapbox/react-native-mapbox-gl": "6.1.2"
"react-native": "0.56.0"
```

**Comment by jussirantala** [comment]
What version of react-native-mapbox-gl you used before? 

- With 6.1.1 and react-native 0.52, I didn't experience this bug
- With 6.1.1 and react-native 0.57.3, the iconSize attribute doesn't work in dev or release
- With 6.1.3 and react-native 0.57.3, the behavior is the same as yours (dev works, release doesnt)
- With 6.1.3 and react-native 0.57.8, the behavior is the same as yours (dev works, release doesnt)

These files have changed between versions 6.1.1 and 6.1.3:
https://pastebin.com/raw/gW9aGZFc

**Comment by jussirantala** [comment]
I found a commit possibly related to this issue. The commit is in versions 6.1.2 and 6.1.3 published in npm.
Anyone experiencing this bug with iOS (the change is Android specific)?

https://github.com/mapbox/react-native-mapbox-gl/commit/d8fb1b8b2d4979d6f8f3a31cc785f1c736aae4fd#diff-01be14636c2db27fdec07f486b439e33

**Comment by AtlantisPleb** [comment]
> I found a commit related to this issue. The commit is in versions 6.1.2 and 6.1.3 published in npm.
> Anyone experiencing this bug with iOS (the change is Android specific)?

Yes, it is happening now for me on iOS with mapbox 6.1.2 and RN 0.57.7. Icon sizes are about 2x bigger in release compared to debug.

**Comment by jussirantala** [comment]
I tried to set iconSize to 1 and edit the original resolution of the image so it will be right size in dev but it's still different size in release build. So it seems the iconSize is not using it's default value (1) in release, this is something else.

**Comment by Darkilen** [comment]
> What version of react-native-mapbox-gl you used before?

I upgraded from 6.1.0 with RN 0.55.2 to 6.1.2 with RN 0.56.0 (worked fine in 6.1.0 with RN 0.55.2)

**Comment by LucasYver** [comment]
Hello,

I have the same issue, do you have a solution ?7

Mapbox 6.1.3
RN 0.57.7

**Comment by Boshoff93** [comment]
+1, experiencing the same issue with RN 0.57.8 and Mapbox 6.1.3

**Comment by jussirantala** [comment]
As a workaround I use 3 times smaller iconSize if not in development environment (I have a environment variable for that).

Not tested in iOS.

**Comment by sethjdev** [comment]
This little helper I made manages the icon sizes relative to their environment and platform. Hope it helps someone else.

`import { Platform } from 'react-native'
import { RELEASE_MODE } from 'react-native-dotenv'

export default base => {

    if (RELEASE_MODE !== "true") return base

    if (Platform.OS === "ios") {
        return base / 3
    } else {
        return base
    }

}`
